nondex-common/src/main/java/edu/illinois/nondex/shuffling/ControlNondeterminism.java

```diff
@@ -93,93 +70,47 @@ private static Random getRandomnessSource(String source) {
         }
 
         List<T> newOrder = Arrays.asList(originalOrder);
-        newOrder = ControlNondeterminism.internalShuffle(newOrder, ControlNondeterminism.getSource());
+
+        newOrder = nondex.permutation(newOrder);
+
         newOrder.toArray(originalOrder);
 
         // return in place
         return originalOrder;
```

Comment:
Well... that method is not part of the NonDex interface. There are two reasons why we don't create a new array:
1) It would require using reflection to find the type of the array I need to create (originalOrder.getClass() and then create some object of that type again via reflection...).
2) it's safe to do so for what we are doing, since all APIs are saying that the user can modify the return value.